### PR TITLE
CPLP1425 & CPLP 1442 Connect partner network to BPDM dev

### PR DESCRIPTION
**Pull request by manojava-gk**

### CODE REVIEW

**Review by lavanya-bmw — DISMISSED** [review]
looks good to me

File: cx-portal-shared-components/src/components/basic/Table/PageLoadingTable.tsx [hunk 1]
```diff
@@ -77,16 +86,59 @@ export const PageLoadingTable = function <T>({
...
+      const result = data.content.map((x: any) => x.legalEntity.bpn)
+      if (additionalHooks) {
+        await additionalHooks
+          .mutationRequest(result)
+          .unwrap()
+          .then((payload: any) => {
+            //new country attribute && member attributes based on the response
+            let finalObj = JSON.parse(JSON.stringify(data?.content))
+            finalObj = addCountryAttribute(finalObj, payload)
+            finalObj = addMemberAttribute(finalObj, additionalHooks.queryData)
```
Comment (oyo) [inline cx-portal-shared-components/src/components/basic/Table/PageLoadingTable.tsx:117]: we must not hard code any data specific attributes into the shared components

File: cx-portal-shared-components/src/components/basic/Table/PageLoadingTable.tsx [hunk 2]
```diff
@@ -77,16 +86,59 @@ export const PageLoadingTable = function <T>({
...
+    } else {
+      const result = [data.bpn]
+      if (additionalHooks) {
+        await additionalHooks
+          .mutationRequest(result)
+          .unwrap()
+          .then((payload: any) => {
+            //update for country attribute && update member info
+            let finalObj = JSON.parse(JSON.stringify(data))
+            finalObj.country = payload[0].legalAddress.country
```
Comment (oyo) [inline cx-portal-shared-components/src/components/basic/Table/PageLoadingTable.tsx:131]: content specific - not in shared components

File: cx-portal-shared-components/src/components/basic/Table/components/Helper/helper.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+export const isQueryDataPresent = (queryData: any) =>
+  queryData && queryData.length > 0
+
+export const isContentPresent = (data: any) => data && data.content
+
+export const addCountryAttribute = (finalObj: any, payload: any) => {
```
Comment (oyo) [inline cx-portal-shared-components/src/components/basic/Table/components/Helper/helper.ts:26]: content specific - not in shared components

File: cx-portal-shared-components/src/components/basic/Table/components/Helper/helper.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    payload.forEach((y: any) => {
+      if (x.legalEntity.bpn === y.legalEntity) {
+        x.legalEntity.country = y.legalAddress.country
+      }
+    })
+  })
+  return finalObj
+}
+
+export const addMemberAttribute = (finalObj: any, queryData: any) => {
```
Comment (oyo) [inline cx-portal-shared-components/src/components/basic/Table/components/Helper/helper.ts:37]: content specific - not in shared components

**Review by oyo — CHANGES_REQUESTED** [review]
I missed that on our last call. Under no circumstances should we hard code any content specific code into shared components. The usual way is for the caller to give a function as a prop that would handle the use case specific stuff.

**Comment by oyo** [comment]
please also update release notes

**Comment by manojava-gk** [comment]
@oyo Have addressed your comments. Let me know your thoughts

**Review by oyo — APPROVED** [review]
lgtm

**Pull request merged**